src/middlewared/middlewared/plugins/reporting/iostat.py

```diff
@@ -0,0 +1,75 @@
+from collections import namedtuple
+import time
+
+DiskStat = namedtuple("DiskStat", ["rd_ios", "rd_bytes", "wr_ios", "wr_bytes", "tot_ticks"])
+
+
+class DiskStats:
+    def __init__(self):
+        self.prev = None
+        self.prev_time = None
+        self.current = None
+        self.current_time = None
+        self._read()
+
+    def _read(self):
+        with open("/proc/diskstats") as f:
+            lines = f.readlines()
+
+        self.current = {}
+        for line in lines:
+            line = line.split()
+
+            disk = line[2]
+
+            if disk.startswith("loop"):
+                continue
+
+            parent = disk
+            while parent and (parent[-1].isdigit() or parent[-1] == "p" and parent.startswith("nvme")):
+                parent = parent[:-1]
+            if parent in self.current:
+                continue
+
+            disk_stat = DiskStat(
+                rd_ios=int(line[3]),
+                rd_bytes=int(line[5]) * 512,
+                wr_ios=int(line[7]),
+                wr_bytes=int(line[9]) * 512,
+                tot_ticks=int(line[12]),
+            )
+
+            self.current[disk] = disk_stat
+
+        self.current_time = time.monotonic()
+
+    def get(self):
+        self.prev = self.current
+        self.prev_time = self.current_time
+        self._read()
+
+        read_ops = 0
+        read_bytes = 0
+        write_ops = 0
+        write_bytes = 0
+        total_ticks = 0
+        for disk, current in self.current.items():
+            prev = self.prev.get(disk)
+            if prev is None:
+                continue
+
+            read_ops += current.rd_ios - prev.rd_ios
+            read_bytes += current.rd_bytes - prev.rd_bytes
+            write_ops += current.wr_ios - prev.wr_ios
+            write_bytes += current.wr_bytes - prev.wr_bytes
+            total_ticks = max(total_ticks, current.tot_ticks - prev.tot_ticks)
```

Comment:
I believe we do not want to know the most busy disk %.
We do want to know an average because odds are all the disks will be in a pool and its likely the ZIL device will be most busy than the rest, which doesn't tell us much of a story. Now if the average % busy is high then it may be a concern.